### DISCUSSION TITLE
[FeatureRequest] Variables: The All option in the variable drop-down should check/uncheck every single value

### DISCUSSION THREAD

**Discussion by AlKult** [grafana/grafana#37013] [Dashboards, Panels & Visualizations]
Hi,

**What would you like to be added**:
1. If All gets checked, every other value in the drop-down should also get checked
2. If All gets unchecked, every checked value should also get unchecked

**Why is this needed**:
Imagine there are 1000 values in the drop-down. If you need to check only 995, in the current implementation of Grafana you'll have manually check all 995 values. Using the inverse logic, you can now check All and then uncheck the 5 values you don't need which will give you the 995 that you do need. Same end result, much less clicking.

The functionality I'm suggesting is identical to how Excel handles it. In Excel, if you have a column with a filter, checking (Select All) will also check every single value in the column, and vice versa unchecking (Select All) will also uncheck every checked value in the column.
The appearance of the tick changes to square when the selection is not full. This is to denote partial selection, which would also be great to see Grafana do.

Best regards,
Albert.

**Comment by AlKult** [comment]
Hi!

Just wondering if anyone's picked this up so far?

Best regards,
Albert.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
